lib/nostrum/struct/snowflake.ex

```diff
@@ -62,4 +62,16 @@ defmodule Nostrum.Struct.Snowflake do
   @spec dump(t) :: external_snowflake
```

Comment:
Sorry for not catching this before, but shouldn't this spec also have a `no_return` clause? Also, thoughts on `!`ing this method since it can throw an error?

Reply:
I've strengthened the `is_snowflake/1` guard to make sure `creation_time/1` always works under the correct conditions.